### DISCUSSION TITLE
Full automated setup and configuration

### DISCUSSION THREAD

**Discussion by k-jell** [requarks/wiki#6194] [Help / Questions] [answered]
Hey, 
I was wondering if it is possible to setup and configure wikijs in a fully automated way. Two things I am specifically interested in are:

1. How to create an initial admin user without the need of the setup wizard in a browser?
2. How to configure an authentication module (Generic Oauth2 in my case) without the admin interface, e.g. using environment variables? 

For example with rocketchat it's possible to overwrite any setting using an environment variable and to configure the authentication that way (authorization endpoint, client id/secret and so on): https://docs.rocket.chat/deploy/rocket.chat-environment-configuration/environment-variables

I'm using the dockerized wikijs (2.5).

I already found https://github.com/requarks/wiki/discussions/4664 and https://github.com/requarks/wiki/discussions/4252 but hoping for some more recent info on the issue.

Would appreciate any help and hints on this!

**Comment by Philipinho** [comment, marked as answer]
There are two options I can think of.
1. Create a create a wiki.js installation and export the already configured database. You can then reuse this database anytime you create a new wiki.js installation.
2. Use browser automation.

**Reply by k-jell** [reply]
> There are two options I can think of.
> 
>     1. Create a create a wiki.js installation and export the already configured database. You can then reuse this database anytime you create a new wiki.js installation.

This is what we did in the end. It works fine and we can use variables for all the dynamic values and substitute them when deploying.
